pkg/sql/opt/props/func_dep.go

```diff
@@ -628,6 +628,23 @@ func (f *FuncDepSet) ReduceCols(cols opt.ColSet) opt.ColSet {
 	var removed opt.ColSet
 	cols = cols.Copy()
```

Comment:
There might be additional gains, if we can avoid copying this set if we use the fast path for every column in the set. We have to make sure that callers are ok getting the same set back. I'll attempt to make this change.